featurebyte/session/spark.py

```diff
@@ -269,26 +270,33 @@ def _process_batch_data(self, data: pd.DataFrame, schema: Schema) -> pd.DataFram
         pd.DataFrame
             Processed data
         """
+        if data.empty:
+            return data
+
         for i, column in enumerate(schema.names):
+            current_type = schema.field(i).type
             # Convert decimal columns to float
-            if schema.field(i).type == pa.float64() and not is_float_dtype(data[column]):
+            if current_type == pa.float64() and not is_float_dtype(data[column]):
                 data[column] = data[column].astype(float)
-            elif isinstance(schema.field(i).type, pa.TimestampType) and not is_datetime64_dtype(
+            elif isinstance(current_type, pa.TimestampType) and not is_datetime64_dtype(
                 data[column]
             ):
                 data[column] = pd.to_datetime(data[column])
+            elif current_type == pa.list_(pa.float64()):
+                # Check if column is string. If so, convert to a list.
+                is_string_series = data[column].apply(lambda x: isinstance(x, str))
+                if is_string_series.any():
```

Comment:
Curious what is this not the case? If only some of the elements are string, is it fine to apply the eval on all elements but not just the elements of string type?

Reply:
hm it should technically apply only when all of them are strings.